sources/imdsclient/src/lib.rs

```diff
@@ -422,6 +474,9 @@ mod error {
         #[snafu(display("Response was not UTF-8: {}", source))]
         NonUtf8Response { source: std::string::FromUtf8Error },
 
+        #[snafu(display("IMDSv2 session token was refreshed."))]
```

Comment:
This message is a little confusing to me.  The error is `UnauthorizedTokenRefreshed`, but the message says the token was refreshed.  Maybe I'm missing some context here.

Reply:
My original line of thinking was the token was refreshed because the query returned a 401, but I went ahead and renamed this to **TokenRefreshed**.